### PR TITLE
Migrate: build-tool use vite replace vue-cli

### PR DESCRIPTION
**Pull request by little-buddy**
| Step           | Status |
| -------------- | ------ |
| serve-web      | ✅      |
| build-web      | todo  |
| electron-build | todo  |
| build-ci      | todo  |

### CODE REVIEW

**Comment by Linkontoask** [comment]
Any progress?

**Comment by little-buddy** [comment]
It’s been a while, but I’m still working 💪🏻

| Step           | Status |
| -------------- | ------ |
| serve-web      | ✅      |
| build-web      | ✅  |
| electron-build | todo  |
| build-ci      | todo  |

**Comment by little-buddy** [comment]
| Step           | Status |
| -------------- | ------ |
| serve-web      | ✅      |
| build-web      | ✅  |
| electron-build | doing  |
| build-ci      | doing  |

**Comment by little-buddy** [comment]
| Step           | Status |
| -------------- | ------ |
| serve-web      | ✅      |
| build-web      | ✅  |
| electron-build | doing  |
| build-ci      | todo |

**Comment by little-buddy** [comment]
| Step           | Status |
| -------------- | ------ |
| serve-web      | ✅      |
| build-web      | ✅  |
| electron-build | ✅  |
| build-ci      | todo  |

**Comment by little-buddy** [comment]
| Step           | Status |
| -------------- | ------ |
| serve-web      | ✅      |
| build-web      | ✅  |
| electron-build | ✅  |
| build-ci      | ✅   |

Now!!! 
`ci` can build on `macos`, but it is not adapted for other platforms. 
Except remove loading box, others workd fine.
🎸🎸🎸

**Comment by Linkontoask** [comment]
> Step	Status
> serve-web	✅
> build-web	✅
> electron-build	✅
> build-ci	✅
> Now!!! `ci` can build on `macos`, but it is not adapted for other platforms. Except remove loading box, others workd fine. 🎸🎸🎸

Later I will try to see if the windows system is normal

File: export-env.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export ELECTRON_MIRROR=https://npmmirror.com/mirrors/electron/
+export ELECTRON_BUILDE_RBINARIES_MIRROR=https://npmmirror.com/mirrors/electron-builder-binaries/
+echo "\033[32m ===================================================== \033[0m"
```
Comment (Linkontoask) [inline export-env.sh]: When will this script be invoked

**Review by Linkontoask — COMMENTED** [review]
cool

File: export-env.sh (hunk 1 above)
Comment (little-buddy) [reply export-env.sh]: only run `yarn mirrori` in China for electron

File: package.json [hunk 2]
```diff
@@ -2,35 +2,38 @@
...
+    "dev:web": "cross-env NODE_ENV=development vite --mode browser",
+    "dev:api": "lerna exec --scope @radishes/api -- yarn run dev",
+    "dev:electron": "concurrently -r \"yarn generate:theme\" \"electron-vite dev --mode electron.dev --outDir dist-electron\"",
+    "build": "cross-env NODE_ENV=production vite build --mode browser",
+    "build:electron": "npx electron-vite build --mode electron.build --outDir dist-electron",
+    "build:electron:win": "node script/build.js --win",
+    "build:electron:linux": "node script/build.js --linux",
+    "build:electron:macos": "node script/build.js --mac",
+    "build:api": "lerna exec --scope @radishes/api -- yarn run build",
+    "pkg": "rm -rf dist-electron && yarn build:electron && npx electron-builder --config .electron-builder.config.js",
```
Comment (Linkontoask) [inline package.json]: windows does not have the rm command, so you need to use rimraf to be compatible with different systems

**Review by Linkontoask — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -2,35 +2,37 @@
...
+    "dev:web": "cross-env NODE_ENV=development vite --mode browser",
+    "dev:api": "lerna exec --scope @radishes/api -- yarn run dev",
+    "dev:electron": "concurrently -r \"yarn generate:theme\" \"electron-vite dev --mode electron.dev --outDir dist-electron\"",
+    "build": "cross-env NODE_ENV=production vite build --mode browser",
+    "build:electron": "npx electron-vite build --mode electron.build --outDir dist-electron",
+    "build:electron:win": "node script/build.js --win",
+    "build:electron:linux": "node script/build.js --linux",
+    "build:electron:macos": "node script/build.js --mac",
+    "build:api": "lerna exec --scope @radishes/api -- yarn run build",
+    "pkg": "npx rimraf dist-electron && yarn build:electron && npx electron-builder --config .electron-builder.config.js",
```
Comment (little-buddy) [inline package.json:20]: fix it @Linkontoask

**Comment by Linkontoask** [comment]
🎉🎉🎉🎉🎉

**Pull request merged**